### PR TITLE
Ports - Amy W

### PR DESCRIPTION
**Pull request by amythetester**
# MediaRanker Revisited
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
---      | ---
What did you need to configure and setup to make OAuth installed and work? | Omniauth gem, a mock auth, and the perform login method for tests. It's also good to require login for the appropriate controller actions.
What areas of Rails app code did you need to create/write/modify in order to change logging in to use OAuth? | I needed to great a login action for user. The user fixtures needed uid, username, and email. Many tests need to be updated and have edge cases for when a user is not logged in. 
What was one controller test you updated? Why did you need to update it? | I updated the controller tests for the User show action. This was needed because a login was needed to get to the show page.
Why did we need to mock OAuth for testing? | Without the mock OAuth, you get a csrf error when the tests try to login using production.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -8,36 +8,58 @@ def show
     render_404 unless @user
   end
 
-  def login_form
+  def login
```
Comment (CheezItMan) [inline app/controllers/users_controller.rb:11]: You don't need this method anymore.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -33,22 +33,18 @@
...
-        <% if @login_user %>
-
-        <li class="nav-item app-header__nav_item">
-          <%= link_to "Logged in as #{@login_user.username}", user_path(@login_user), class: "btn btn-primary" %>
-        </li>
-        <li class="nav-item app-header__nav_item">
-          <%= link_to "Log Out", logout_path, method: :post, class: "btn btn-primary" %>
-        </li>
-
+        <% if session[:user_id] %>
```
Comment (CheezItMan) [inline app/views/layouts/application.html.erb:36]: This should instead be `if @login_user`.  Just if the session variable in the cookie doesn't match the user's id in the database.

File: test/controllers/works_controller_test.rb [hunk 3]
```diff
@@ -189,19 +220,64 @@
 
   describe "upvote" do
     it "redirects to the work page if no user is logged in" do
-      skip
+      work_id = works(:poodr)
+
+      expect {
+        post upvote_path(work_id)
+      }.wont_change "Vote.count"
+
```
Comment (CheezItMan) [inline test/controllers/works_controller_test.rb:228]: You should also test that there's a flash notice as well.

File: test/controllers/works_controller_test.rb [hunk 4]
```diff
@@ -189,19 +220,64 @@
...
+      must_respond_with :redirect
+      must_redirect_to work_path(work_id)
+    end
+
+    it "redirects to the work page if a logged in user tries to upvote a work that does not exist" do
+      perform_login
+      work_id = 3490
```
Comment (CheezItMan) [inline test/controllers/works_controller_test.rb:274]: Just for note, it's likely that this ID doesn't exist.  It's not guaranteed.  Instead I suggest using -1, or deleting an existing work and then trying to upvote it.

**Comment by CheezItMan** [comment]
# MediaRanker Revisted
## What We're Looking For



Feature | Feedback
---     | ---
**General** |
Completed OAuth |   Check
Appropriate Tests on WorksController |   Check, with one note
Appropriate Tests on UsersController |  Check
Basic Authorization (who can see what) |  Check
**Overall** |  Nice work, you hit the learning goals for this project.  One note on why we mocked OAuth.  We did so to remove github as a dependency more than the CSRF error.
